### PR TITLE
Moving uninstall and cancel functions to Registry prototype

### PR DESCRIPTION
**Pull request by joeldenning**

### CODE REVIEW

File: src/loader.js [hunk 1]
```diff
@@ -252,6 +224,32 @@
...
+  // 4.4.5
+  Registry.prototype.uninstall = function(key) {
+    if (typeof this !== 'object')
+      throw new TypeError('registry must be an object');
+    var entry = this._registry.registryData[key];
+    if (!entry)
+      throw new TypeError('Module with key ' + key + 'does not exist');
+    var stageEntry = getCurrentStage(entry);
+    if (stageEntry.stage !== 'link' && stageEntry.stage !== 'ready')
+      throw new TypeError('Module is still loading');
```
Comment (guybedford) [inline src/loader.js]: `'Module ' + key + ' is still loading`?

File: src/loader.js [hunk 2]
```diff
@@ -252,6 +224,32 @@
     };
   }
 
+  // 4.4.5
+  Registry.prototype.uninstall = function(key) {
+    if (typeof this !== 'object')
+      throw new TypeError('registry must be an object');
```
Comment (guybedford) [inline src/loader.js]: Capital `R`?

**Comment by guybedford** [comment]
:+1:

**Comment by joeldenning** [comment]
Made those changes.

File: src/loader.js [hunk 3]
```diff
@@ -252,6 +224,32 @@
...
+  // 4.4.5
+  Registry.prototype.uninstall = function(key) {
+    if (typeof this !== 'object')
+      throw new TypeError('Registry must be an object');
+    var entry = this._registry.registryData[key];
+    if (!entry)
+      throw new TypeError('Module ' + key + ' does not exist');
+    var stageEntry = getCurrentStage(entry);
+    if (stageEntry.stage !== 'link' && stageEntry.stage !== 'ready')
+      throw new TypeError('Module is still loading');
```
Comment (guybedford) [inline src/loader.js]: It still feels like the module key would be useful to report here?

Comment (joeldenning) [inline src/loader.js]: Ah I misread which line your previous comment was on, and so I updated the wrong line. Will fix

**Comment by guybedford** [comment]
Thanks!

**Pull request merged**